src/Tasks.UnitTests/MSBuild_Tests.cs

```diff
@@ -1408,5 +1406,44 @@ public void NonexistentTargetFailsAfterSkipped()
             // 2nd invocation of t_missing should fail the build resulting in target not found error (MSB4057)
             logger.FullLog.ShouldContain("MSB4057");
         }
+
+        [Fact]
+        public void MSBuildTaskPassesTaskIdToSpawnedBuilds()
```

Comment:
My last request: add a test for the `IBuildEngine.BuildProjectFile` case :)

Reply:
Done! 🎄